podcast/2.15.md

```diff
@@ -0,0 +1,1299 @@
+
+**Beyang Liu:**
+
+Yeah, of course.
+
+**Sugu Sougoumarane:**
+
+And there is nothing proprietary. And so we are going to open source this. And so we decided to open source it. So Vitess, actually, I don't know how many rules of Google we broke by doing what we did, but we created a sub-version repository in our home directory, and checked in all the code into that repository.
+
+**Beyang Liu:**
+
+Just on your local machine.
+
+**Sugu Sougoumarane:**
+
+The home directory was a mounted drive. So it was a Google mounted drive network drive. So it was on a network drive. It was not checked into Google's monorepo.
+
+**Beyang Liu:**
+
+You took it out of the monorepo.
+
+**Sugu Sougoumarane:**
+
+It was not in the monorepo. It was developed all in our home directory.
+
+**Beyang Liu:**
+
+Got it.
+
+**Sugu Sougoumarane:**
+
+Because we had-
+
+**Beyang Liu:**
+
+Oh, you did that intentionally, because it would be easier to kind of-
+
+**Sugu Sougoumarane:**
+
+So that it was easier to open source it. Initially, we didn't want to start off open sourcing.
+
+So we developed it in the directory and we deployed it within YouTube. So our deployment scripts used to read off of our home directory.
+
+**Beyang Liu:**
+
+And no one called you on that? You were just able to kind of get away with that?
+
+**Sugu Sougoumarane:**
+
+It was okay. Well, the fact of the matter is we were the callers. So we were the enforcers.
+
+**Beyang Liu:**
+
+It was all in the family.
+
+**Sugu Sougoumarane:**
+
+It was all in the family. So it was well-understood and Google's home directory is pretty secure that way.
+
+**Beyang Liu:**
+
+Yeah. That's awesome.
+
+**Sugu Sougoumarane:**
+
+And even when we went to the Google open source team and told them this, they were not shocked or anything. They said, "Yeah, sure." It's no different from you developing some code. So yeah, because it was backed up, it was secured, we were not going to lose that file system. It had all the security, so there was no fear of...
+
+**Beyang Liu:**
+
+I like that. So, give me a sense of timeline. When did Vitess as a project... when was the beginning of that, and then when was it open-sourced for the first time?
+
+**Sugu Sougoumarane:**
+
+So I believe we started the project in 2010. It went into production in 2011, and we opened sourced it in 2012.
+
+**Beyang Liu:**
+
+2012, okay, got it.
+
+**Sugu Sougoumarane:**
+
+2012 I think is when we first open sourced it. So initially we had named it Voltron.
+
+**Beyang Liu:**
+
+Voltron. Is that a Pokémon or-
+
+**Sugu Sougoumarane:**
+
+It's a Japanese cartoon. I think it is something where five different pieces come together to form this big giant thing. And the idea was that Vitess is a bunch of pieces that come together to. So that model still makes sense because Vitess is a distributed system where a bunch of components come and work together. So we named it Voltron, and obviously we could not open source with that name.
```

Comment:
In "And the idea was that Vitess is a bunch of pieces that come together to" should that be "too" or is it a sentence that trails off?